test/serializer/basic/StrictComplex.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright 2004-present Facebook. All Rights Reserved.
+ */
+/**
+ * Copyright 2004-present Facebook. All Rights Reserved.
+ */
+(function() {
+    var isStrict = function() {
+        "use strict";
```

Comment:
I think the testcase should have the following in comment to test out the expectation which will fail. 
```
// does contain:"use strict" 
```
Then you can investigate and fix it.(Culprit logic is in ResidualFunctions.js)

Reply:
I just realize the code itself already tests the strictness in the generated code so that testcase would fail. Now you just need to investigate and fix the failure :-)